### PR TITLE
Use Salsa framework during release to increase the supply-chain security

### PR DESCRIPTION
**Pull request by jkremser**
This is how it looks like on my other repo - https://github.com/jkremser/log2rbac-operator/actions/runs/3234735006#summary-8846153907

it relies on 3 new repo secrets: `COSIGN_PRIVATE_KEY`, `COSIGN_PUBLIC_KEY` and `COSIGN_PASSWORD`
`COSIGN_PASSWORD` is a passphrase for the private key. The private key is used for signing the artifacts (container images, provenance files, sboms, checksums). The public key is also part of the repo and is used for verifying the signatures.

During the release we push the signatures of container images into container registries so that one can verify the validity of the image. Also some artifacts are published together with the release, check the Assets section on [log2rbac release](https://github.com/jkremser/log2rbac-operator/releases/tag/v0.0.5) page.

inspired / kudos to https://github.com/philips-labs/slsa-provenance-action/blob/main/.github/workflows/ci.yaml

Signed-off-by: Jirka Kremser <jiri.kremser@gmail.com>

### CODE REVIEW

**Review by kuritka — APPROVED** [review]
Lgtm

File: cosign.pub [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-----BEGIN PUBLIC KEY-----
```
Comment (somaritane) [inline cosign.pub:1]: just  thinking, what should be the procedure for cosign key pair regeneration

**Review by somaritane — CHANGES_REQUESTED** [review]
@jkremser looks awesome! 
I suggest reflecting these changes in the dev documentation, also wrt the case when cosign key pair should be regenerated.

**Review by ytsarev — APPROVED** [review]
@jkremser it looks really really good. Is it possible to have brief contextual documentation for the implementation? Fine to have it as a separate PR.

**Review by ytsarev — APPROVED** [review]

**Review by kuritka — APPROVED** [review]

**Comment by jkremser** [comment]
@somaritane I've added the docs, can you "ptal" again?

**Review by somaritane — APPROVED** [review]
@jkremser lgtm now, thank you!

**Pull request merged**